firestore/FirestoreSwiftUIExample/ViewModels/RestaurantViewModel.swift

```diff
@@ -35,6 +35,26 @@ class RestaurantViewModel: ObservableObject {
     unsubscribe()
   }
 
+  func add(review: Review) {
+    do {
+      _ = try restaurant.reference!.collection("ratings").addDocument(from: review)
```

Comment:
same type issue as before. I still haven't tried implementing `FirestoreValue<T: Codable, Identifiable>` to handle this, but it's still on my radar to look at at some point

Reply:
No problem 👍 

If you want to reduce the amount of string literals, you can define an extension on the Restaurant type.

```swift
extension Restaurant {
  var ratingsCollection: CollectionReference? {
    return reference?.collection("ratings")
  }
}
```